### PR TITLE
Update Service Catalog REST APIs and fix issues

### PR DESCRIPTION
**Pull request by vithu30**
This PR contains following changes:

- Add examples to newly added resources in Publisher REST API and Service Catalog REST API
- Increase column length of serviceKey to 512
- Fix issue in export service implementation
- Not allowing user to update name of existing service
- Fix issues in adding and updating services via `POST /services` and `PUT /services/{serviceId}`

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml [hunk 1]
```diff
@@ -18,7 +18,79 @@ info:
...
+    "scope": "service_catalog:service_view service_catalog:service_write",
+    "token_type": "Bearer",
+    "expires_in": 3600
+    }
+    ```
+    Now you have a valid access token, which you can use to invoke an API.
+    Navigate through the API descriptions to find the required API, obtain an access token as described above and invoke the API with the authentication header.
+    If you use a different authentication mechanism, this process may change.
+
+  version: '1.0'
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml:93]: 0.9?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml [hunk 2]
```diff
@@ -41,12 +113,34 @@ paths:
...
+        ```
+        curl -k -X GET 'https://127.0.0.1:9443/api/am/service-catalog/v0/services?version=1.0.0' -H
+        'Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8'
+        ```
+
+        `Get all versions of a given service`
+        ```
+        curl -k -X GET 'https://127.0.0.1:9443/api/am/service-catalog/v0/services?name=%22Pet%20Service%22' -H
+        'Authorization: Bearer ae4eae22-3f65-387b-a171-d37eaa366fa8'
+        ```
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml:143]: Do we have any other way to add multiple examples in x-code-samples?

**Review by malinthaprasan — APPROVED** [review]

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml (hunk 1 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml]: Fixed

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml (hunk 2 above)
Comment (vithu30) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.service.catalog/src/main/resources/service-catalog-api.yaml]: Yes, fixed in 6091a00.

**Comment by malinthaprasan** [comment]
There are below failures which are not related to the feature. (the same tests seems to fail in other PRs as well. We need to look into this separately.)

```
2021-03-04T20:02:50.9464972Z INFO  [org.wso2.carbon.automation.extensions.servers.carbonserver.CarbonServerManager] - Jacoco coverage HTML report path : /home/runner/work/carbon-apimgt/product-apim/modules/integration/tests-integration/tests-backend/target/jacoco/coverage/index.html
2021-03-04T20:02:51.0598933Z [ERROR] Tests run: 656, Failures: 2, Errors: 0, Skipped: 0, Time elapsed: 12,089.958 s <<< FAILURE! - in TestSuite
2021-03-04T20:02:51.0601453Z [ERROR] testOperationalLevelOAuthScopesForGraphql(org.wso2.am.integration.tests.graphql.GraphqlTestCase)  Time elapsed: 11.877 s  <<< FAILURE!
2021-03-04T20:02:51.0603605Z java.lang.AssertionError: Response code is not as expected expected [403] but found [200]
2021-03-04T20:02:51.0606662Z 	at org.wso2.am.integration.tests.graphql.GraphqlTestCase.testOperationalLevelOAuthScopesForGraphql(GraphqlTestCase.java:286)
2021-03-04T20:02:51.0609089Z 
2021-03-04T20:02:51.0611190Z [ERROR] testOperationalLevelOAuthScopesForGraphql(org.wso2.am.integration.tests.graphql.GraphqlTestCase)  Time elapsed: 11.812 s  <<< FAILURE!
2021-03-04T20:02:51.0652785Z java.lang.AssertionError: Response code is not as expected expected [403] but found [200]
2021-03-04T20:02:51.0656135Z 	at org.wso2.am.integration.tests.graphql.GraphqlTestCase.testOperationalLevelOAuthScopesForGraphql(GraphqlTestCase.java:286)
2021-03-04T20:02:51.0659717Z 
2021-03-04T20:02:51.4343764Z [INFO] 
2021-03-04T20:02:51.4344700Z [INFO] Results:
2021-03-04T20:02:51.4345255Z [INFO] 
2021-03-04T20:02:51.4345816Z [ERROR] Failures: 
2021-03-04T20:02:51.4352056Z [ERROR] org.wso2.am.integration.tests.graphql.GraphqlTestCase.testOperationalLevelOAuthScopesForGraphql(org.wso2.am.integration.tests.graphql.GraphqlTestCase)
2021-03-04T20:02:51.4356161Z [ERROR]   Run 1: GraphqlTestCase.testOperationalLevelOAuthScopesForGraphql:286 Response code is not as expected expected [403] but found [200]
2021-03-04T20:02:51.4358834Z [ERROR]   Run 2: GraphqlTestCase.testOperationalLevelOAuthScopesForGraphql:286 Response code is not as expected expected [403] but found [200]
2021-03-04T20:02:51.4363120Z [INFO] 
2021-03-04T20:02:51.4363617Z [INFO] 
```

**Pull request merged**